azurelinuxagent/common/event.py

```diff
@@ -631,26 +629,24 @@ def collect_events(self): # pylint: disable=R0914
                         else:
                             self._update_legacy_agent_event(event, event_file_creation_time)
 
-                    event_list.events.append(event)
+                    enqueue_event_func(event)
```

Comment:
Changed the logic to more on-demand approach. Instead of using a list to capture all events and then send them, sending them out to wireserver as soon as they're captured. This on-demand mechanism would help us salvage some extra time if some event file takes longer to parse